_sources/rust-2019-one-zero-is-a-beautiful-version.md

```diff
@@ -0,0 +1,25 @@
+---
+title: "1.0 is a beautiful version"
+---
+
+# Rust 2019 (5): 1.0 is a beautiful version
+
+When I wrote last years post, [Rust 2018: The year we learn Rust 1.0](https://yakshav.es/rust-2018),
+I tried to make an argument to focus on ecosystem improvements over language improvements.
+I specifically called out 2 spaces: a stable http client and documented async ecosystem.
+
+1 year in, the situation is the same: we still don't have a 1.0 HTTP client in our ecosystem. While `hyper` and `request` are still the go-to libraries and good, they are still not committed to stability. Also, no other library has emerged to fill that hole. The more our ecosystem grows, the larger the community-wide cost of following breaking changes in these libraries will get.
+
+Proper, high-level documentation of async/futures was pushed tied to the arrival of `async/await`, which was targeted for the edition release and later shifted back. I like `async/await`, I still think the decision to tie documentation to it was a mistake.
+
+This has a ripple effect: everything based on these libraries and features will hold back committing to stability: web and API frameworks, other network protocol implementations or even just plain tutorials.
```

Comment:
btw, my opinion: forcing futures.rs into core will lead to fragmentation because it doesn't work for embedded.

Reply:
This is a side discussion that I didn't necessarily want to touch and doesn't really impact my point, but hasn't there been a lot of work done to avoid that, by e.g. not pushing `executor` into core, just `Waker`?

Reply:
Won't help, there's alot more details to it that would be easy to resolve. Problem is, the debate has degraded to reddit style. People just repeat the same argument so it stays on top.

Reply:
Well, I'm seriously not equipped for that discussion, so I do not plan to touch it there, but it seems to be that there are two opposing, but no tangible artifact for a technical discussion.